modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -170,6 +170,10 @@ object CurrencySnapshotCreator {
 
           (height, subHeight) <- getHeightAndSubHeight(lastArtifact, deprecated, remainedActive, accepted)
 
+          feeTransactions = dataApplicationAcceptanceResult
+            .map(result => SortedSet.from(result.feeTransactions))
+            .filter(_.nonEmpty)
```

Comment:
You need to consider whether in the empty case `feeTransactions` field should be a `None` or `Some(SortedSet.empty)`. The none case means that this field is not yet used and some with empty set means that it's used but there are no fees to be paid. Not saying that we can't put None in (maybe it's interchangeable), but the semantic difference is there and it will be reflected in the hash.

Reply:
@TheMMaciek That's what I had yesterday but `Some(SortedSet.empty)` causes the github tests to fail. If that's the correct implementation then we need to adjust the tests?

Reply:
I guess, `Some(SortedSet.empty)` should work in my opinion as it's a valid scenario in my opinion.

Reply:
@TheMMaciek I resolved it the same way it's handled for the `messages` field.